### PR TITLE
MM-10028 Add system notice feature with API deprecation notice

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Adds the system notice feature and includes the API v3 deprecation notice. New notices can be added by modifying `components/system_notice/notices.jsx`.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-10028

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates

### CODE REVIEW

**Comment by jwilander** [comment]
@jasonblais it's ready to test but I'll need @asaadmahmood's help with a few things:

* Making sure my CSS is all good
* Handling themes with the CSS
* The spec had a blue MM icon, but I didn't have it so I just used a black one we already had

**Comment by jasonblais** [comment]
Sorry, just noticed this is still in PM review.

@jwilander @asaadmahmood moving to dev review. Can one of you help investigate the build failure?

File: i18n/en.json [hunk 1]
```diff
@@ -1,4 +1,8 @@
 {
+  "system_notice.title": "<strong>System Message</strong><br>from Mattermost",
```
Comment (saturninoabril) [inline i18n/en.json]: Maybe `<br>` is not necessary here?

File: actions/views/notice.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
```
Comment (saturninoabril) [inline actions/views/notice.js]: Should be 2015-present?

File: i18n/en.json (hunk 1 above)
Comment (jwilander) [reply i18n/en.json]: Yeah it's needed on the next line and I didn't want to split it up into two different messages

File: actions/views/notice.js (hunk 2 above)
Comment (jwilander) [reply actions/views/notice.js]: I thought it was supposed to be the year it was created

Comment (saturninoabril) [reply actions/views/notice.js]: Thanks for the reminder. I was confused about the latest agreement.

File: i18n/en.json (hunk 1 above)
Comment (saturninoabril) [reply i18n/en.json]: 👍 (I just noticed that it's not the same as https://github.com/mattermost/mattermost-webapp/pull/1118/files#diff-08e5c13a7cc7c4fdc5ec0c7847bc7679R24)

**Pull request merged**